### ISSUE TITLE
Crashing on opening "Files" section

### ISSUE THREAD

**Issue by thedaftbadger** [mrvautin/openKB#6]
Clicking files option in menu causes server to crash...

![screenshot_090315_115508_am](https://cloud.githubusercontent.com/assets/13657363/9656751/c03a41d0-5232-11e5-801f-f05033bd1c9f.jpg)

```
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'restart' ]
2 info using npm@2.11.3
3 info using node@v0.12.7
4 verbose run-script [ 'prerestart',
4 verbose run-script   'prestop',
4 verbose run-script   'stop',
4 verbose run-script   'poststop',
4 verbose run-script   'restart',
4 verbose run-script   'prestart',
4 verbose run-script   'start',
4 verbose run-script   'poststart',
4 verbose run-script   'postrestart' ]
5 info prerestart openkb@0.0.8
6 info prestop openkb@0.0.8
7 info stop openkb@0.0.8
8 info poststop openkb@0.0.8
9 info restart openkb@0.0.8
10 info prestart openkb@0.0.8
11 info start openkb@0.0.8
12 verbose unsafe-perm in lifecycle true
13 info openkb@0.0.8 Failed to exec start script
14 verbose stack Error: openkb@0.0.8 start: `node app.js`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:213:16)
14 verbose stack     at EventEmitter.emit (events.js:110:17)
14 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
14 verbose stack     at ChildProcess.emit (events.js:110:17)
14 verbose stack     at maybeClose (child_process.js:1015:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
15 verbose pkgid openkb@0.0.8
16 verbose cwd c:\xampp\htdocs\openkb
17 error Windows_NT 6.3.9600
18 error argv "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "restart"
19 error node v0.12.7
20 error npm  v2.11.3
21 error code ELIFECYCLE
22 error openkb@0.0.8 start: `node app.js`
22 error Exit status 1
23 error Failed at the openkb@0.0.8 start script 'node app.js'.
23 error This is most likely a problem with the openkb package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     node app.js
23 error You can get their info via:
23 error     npm owner ls openkb
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by mrvautin** [comment]
Fixed in latest commit. Was some image dimension checking. If file is not an image it dies. Not sure why that was needed but removed now. Also allows for any files to be uploaded and not just images.

**Comment by mrvautin** [comment]
Closing. Will reopen if not fixed.

**Comment by thedaftbadger** [comment]
All fixed. One thing that could be good is if it read sub folders on the file management. The ability to create folders from the web would be useful if it recognised sub-folders.

I have a lot of articles created now (for user instruction) so I have organised the images folder with sub-folders.

If not it's no problem I have an easy workflow for adding images into the public folder but it might be easier this way for others.

**Comment by mrvautin** [comment]
Subfolders should automatically work when reading in the files. Creating the ability to upload into subfolders might be possible at some point.

**Comment by thedaftbadger** [comment]
Files within sub-folders are visible and shown I just wondered about seeing the actual sub-folders in the "Files" section.

**Comment by mrvautin** [comment]
I'm not really sure what you mean. Are you able to explain further/screenshot?

**Comment by thedaftbadger** [comment]
Sure.... The files section now only lists files such as jpg's. But I have folders in the image root. It would be good if the files section of the app completely mirrored the images folder showing any file structure as well as jpg's etc.

When you have a lot of articles sorting the images folder is kind of necessary.

**Comment by mrvautin** [comment]
I'm still unsure. Here is mine currently. There is a subfolder with a mixture of images and PDF's. I can see the ordering is a little off but it shows all files.

![image](https://cloud.githubusercontent.com/assets/299398/9677868/13ba0156-531e-11e5-9c5f-f4b49fa47c6f.png)

**Comment by thedaftbadger** [comment]
Mine shows all files which is good but it would be nice if you could create sub folders and view sub folders on the app.

**Comment by mrvautin** [comment]
Fixed in 4c666b1efb374ff7e97cba7860eb12eada9c63bc.
